src/main/java/uk/ac/ebi/atlas/experimentpage/ExternallyAvailableContentController.java

```diff
@@ -0,0 +1,86 @@
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static uk.ac.ebi.atlas.utils.GsonProvider.GSON;
+
+@Controller
+public class ExternallyAvailableContentController {
+    private final ExpressionAtlasContentService expressionAtlasContentService;
+
+    public ExternallyAvailableContentController(ExpressionAtlasContentService expressionAtlasContentService) {
+        this.expressionAtlasContentService = expressionAtlasContentService;
+    }
+
+    /*
+    I could be nicer and maybe even have tests:
+    - the "View in Array Express" needs an independent URL
+    - the QC report needs a redirect to another Atlas page, preserving the access key
+    - the typical resource needs to circle back to this page
+    */
+    private JsonObject contentAsJson(ExternallyAvailableContent content, String accession, String accessKey) {
+        JsonObject result = content.description.asJson();
+        if ("redirect".equals(content.uri.getScheme())) {
+            try {
+                result.addProperty("url", new URL(content.uri.getSchemeSpecificPart()).toExternalForm());
+
+            } catch (MalformedURLException e) {
+                result.addProperty("url",
+                        MessageFormat.format("{0}{1}",
+                                content.uri.getSchemeSpecificPart(),
+                                isNotEmpty(accessKey) ? "?accessKey=" + accessKey : "")
+                );
+            }
+
+        } else {
+            result.addProperty(
+                    "url",
+                    MessageFormat.format("experiments-content/{0}/resources/{1}{2}",
+                            accession, content.uri.toString(), isNotEmpty(accessKey) ? "?accessKey=" + accessKey : ""
+            ));
+        }
+        return result;
+    }
+
+    private JsonArray contentAsJson(List<ExternallyAvailableContent> contents, String accession, String accessKey) {
+        JsonArray result = new JsonArray();
+        for (ExternallyAvailableContent content: contents) {
+            result.add(contentAsJson(content, accession, accessKey));
+        }
+        return result;
+    }
+
+    @ResponseBody
+    @RequestMapping(value = "json/experiments/{experimentAccession}/resources/{contentType}",
+                    method = RequestMethod.GET,
+                    produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public String list(@PathVariable String experimentAccession,
+                       @PathVariable String contentType,
+                       @RequestParam(value = "accessKey", defaultValue = "") String accessKey) {
+        return GSON.toJson(
+                contentAsJson(
+                        expressionAtlasContentService.list(
+                                experimentAccession,
+                                accessKey,
+                                ExternallyAvailableContent.ContentType.valueOf(contentType)),
+                        experimentAccession,
+                        accessKey));
+    }
+
```

Comment:
extra space alert 🕵️

Reply:
I check the style with `atlas` rules, but there is no problems in the files. What rules are you using?

Reply:
Which `atlas` rules are you following? Can you please let me know the link to it? I did not check with any rules it just looks weird an extra space between `if-else` blocks and closing curly braces.